src/main/java/edu/stanford/snap/spinn3rHadoop/utils/DocumentFilter.java

```diff
@@ -205,6 +212,18 @@ else if(removeUnparsableURL){
 			return false;
 		}
 		
+		/** Clean content if cleanContent is set
+		 * 
+		 */
+		if (contentCleaner != null) {
+			if (contentCleaner.equals(ParseCLI.CONTENT_CLEANER_INCLUDE_SPECUAL_CHAR)) {
+				ContentCleaner.cleanContent(d, true);
+			}
+			else {
+				ContentCleaner.cleanContent(d);
+			}
+		}
+		
```

Comment:
Since this is costly operation, could we move it a bit further down the list? Maybe even all the way till the end, or at least to just before `Filter by content` section. Hopefully, doing so we could avoid unnecessary computation if other conditions are not satisfied.